modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -129,6 +129,63 @@ G_TYPED_KERNEL(GBackgroundSubtractor, <GMat(GMat, BackgroundSubtractorParams)>,
     }
 };
 
+/** @brief Structure for the Kalman filter's initialization parameters.*/
+
+struct GAPI_EXPORTS KalmanParams
+{
+    // initial state
+
+    //! corrected state (x(k)): x(k)=x'(k)+K(k)*(z(k)-H*x'(k))
+    Mat state;
+    //! posteriori error estimate covariance matrix (P(k)): P(k)=(I-K(k)*H)*P'(k)
+    Mat errorCov;
+
+    // dynamic system description
+
+    //! state transition matrix (A)
+    Mat transitionMatrix;
+    //! measurement matrix (H)
+    Mat measurementMatrix;
+    //! process noise covariance matrix (Q)
+    Mat processNoiseCov;
+    //! measurement noise covariance matrix (R)
+    Mat measurementNoiseCov;
+    //! control matrix (B) (Optional: not used if there's no control)
+    Mat controlMatrix;
+};
+
+void checkParams(const KalmanParams& kfParams, const GMatDesc& measurement,
+                        const bool controlCapable, const int type);
+
+G_TYPED_KERNEL(GKalmanFilter, <GMat(GMat, GOpaque<bool>, GMat, KalmanParams)>,
+               "org.opencv.video.KalmanFilter")
+{
+    static GMatDesc outMeta(const GMatDesc& measurement, const GOpaqueDesc&,
+                            const GMatDesc& control, const KalmanParams& kfParams)
+    {
+        int type = kfParams.transitionMatrix.type();
+        int depth = (type == CV_32FC1) ? CV_32F : CV_64F;
+
+        checkParams(kfParams, measurement, true, type);
+
+        GAPI_Assert(!control.size.empty() && control.depth == depth &&
+                    control.chan == 1);
+
+        GAPI_Assert(control.size.height == kfParams.controlMatrix.cols &&
+                    control.size.width == 1);
```

Comment:
Please move all the asserts to the check function

Reply:
+1. I suppose you mean `checkParams` should have another flag like `haveControl` and these checks should be under the `if`.

Reply:
Could you tell me please how you suggest passing control when there are no in the system? Create empty GMatDesc?

Reply:
May be. Or make this argument last and with the default value `GMatDesc()` to avoid passing empty GMatDesc in the overload without control

Reply:
I think this is not a very good style. @rganov please comment

Reply:
Done.